skbio/tree/_tree.py

```diff
@@ -3269,3 +3269,295 @@ def unpack_by_func(self, func):
+
+        Raises
+        ------
+        NoLengthError
+            A NoLengthError will be thrown if a node without length is
+            encountered
+
+        See Also
+        --------
+        distance
+        tip_tip_distances
+
+        Examples
+        --------
+        >>> from six import StringIO
+        >>> from skbio import TreeNode
+        >>> # example for a non-ultrametric tree
+        >>> tree = TreeNode.read(["((a:1,b:2)c:3,(d:4,e:5)f:6)root;"])
+        >>> tree.is_ultrametric()
+        False
+        >>> tip_root = tree.root()
+        >>> print([tip_root.distance(tree.find(tip)) for tip in tree.tips()])
+        [4.0, 5.0, 10.0, 11.0]
+        >>> # example for an ultrametric tree
+        >>> tree = TreeNode.read(['((a:3,b:3)e:7,(c:2,d:2)f:8);'])
+        >>> tree.is_ultrametric()
+        True
+        >>> tip_root = tree.root()
+        >>> print([tip_root.distance(tree.find(tip)) for tip in tree.tips()])
+        [10.0, 10.0, 10.0, 10.0]
+        """
+        tip_root = self.root()
+        metrics = set()
+        for tip in self.tips():
+            root_to_tip_dist = tip_root.distance(self.find(tip))
+            metrics.add(root_to_tip_dist)
+        if len(metrics) > 1:
+            return False
+        return True
+
+    @experimental(as_of="0.5.3")
+    def scale_length(self):
+        """Returns the tree where the branch lengths are
+        scaled, i.e. divided by (i) the root-to-tip distance
+        for ultrametric trees and (ii) the total branch
+        length (method `descending_branch_length`) for
+        non-ultrametric trees.
+
+        Returns
+        -------
+        TreeNode
+            The length-scaled tree
+
+        Raises
+        ------
+        NoLengthError
+            A NoLengthError will be thrown if a node without length is
+            encountered (from the `distance` method)
+
+        See Also
+        --------
+        distance
+        is_ultrametric
+
+        Examples
+        --------
+        >>> from six import StringIO
+        >>> from skbio import TreeNode
+        >>> # scaling non-ultrametric tree uses total branch length
+        >>> t = TreeNode.read(["((a:1,b:2)c:3,d:4)root;"])
+        >>> t2 = t.scale_length()
+        >>> [(n.length, t2.find(n.name).length) for n in t.preorder(False)]
+        [(3.0, 0.3), (1.0, 0.1), (2.0, 0.2), (4.0, 0.4)]
+        >>> # scaling ultrametric tree uses root-to-tip
+        >>> t = TreeNode.read(['((a:3,b:3)e:7,f:10)root;'])
+        >>> t2 = t.scale_length()
+        >>> [(n.length, t2.find(n.name).length) for n in t.preorder(False)]
+        [(7.0, 0.7), (3.0, 0.3), (3.0, 0.3), (10.0, 1.0)]
+        """
+        tcopy = self.deepcopy()
+        if tcopy.is_ultrametric():
+            # if tree is ultrametric, use longest edge
+            tip_root = tcopy.root()
+            for tip in tcopy.tips():
+                ref_branch_length = tip_root.distance(tcopy.find(tip))
+                break
```

Comment:
I don't understand this for loop?! Why do you break, after the first iteration? See above about use of `.find`. Are you looking for longest root tip distance?